File: actionbar/src/main/java/mobi/cangol/mobile/actionbar/internal/ActionBarImpl.java
package mobi.cangol.mobile.actionbar.internal;

import android.view.View;
import android.view.View.OnClickListener;

import java.util.ArrayList;

import mobi.cangol.mobile.actionbar.ActionBar;
import mobi.cangol.mobile.actionbar.ActionMenu;
import mobi.cangol.mobile.actionbar.ActionMenuItem;
import mobi.cangol.mobile.actionbar.ActionMode;
import mobi.cangol.mobile.actionbar.ActionMode.Callback;
import mobi.cangol.mobile.actionbar.ActionTab;
import mobi.cangol.mobile.actionbar.ActionTabItem;
import mobi.cangol.mobile.actionbar.OnNavigationListener;
import mobi.cangol.mobile.actionbar.view.ActionBarView;

/**
 * @author Cangol
 */
public final class ActionBarImpl extends ActionBar {
    private ActionBarView mActionBarView;

    public ActionBarImpl(ActionBarView view) {
        this.mActionBarView = view;
    }

    @Override
    public void setCustomHomeAsUpIndicator(int homeId, int upId) {
        mActionBarView.setCustomHomeAsUpIndicator(homeId, upId);
    }

    @Override
    public void setDisplayShowHomeEnabled(boolean show) {
        mActionBarView.setDisplayShowHomeEnabled(show);
    }

    @Override
    public void displayIndicator(float slideOffset) {
        mActionBarView.displayIndicator(slideOffset);
    }

    @Override
    public void displayHomeIndicator() {
        mActionBarView.displayHomeIndicator();
    }

    @Override
    public void displayUpIndicator() {
        mActionBarView.displayUpIndicator();
    }

    @Override
    public void setIndicatorColor(int color) {
        mActionBarView.setIndicatorColor(color);
    }

    @Override
    public void setLeftMenu(int id, int text, int icon, OnClickListener listener) {
        mActionBarView.setLeftMenu(id,text,icon,listener);
    }

    @Override
    public void clearLeftMenu() {
        mActionBarView.clearLeftMenu();
    }

    @Override
    public boolean onBackPressed() {
        return mActionBarView.onBackPressed();
    }

    public void setBackgroundColor(int color) {
        mActionBarView.setBackgroundColor(color);
    }

    public void setBackgroundResource(int resId) {
        mActionBarView.setBackgroundResource(resId);
    }

    @Override
    public boolean isShow() {
        return mActionBarView.getVisibility() == View.VISIBLE;
    }

    @Override
    public void setShow(boolean show) {
        mActionBarView.setVisibility(show ? View.VISIBLE : View.GONE);

    }

    @Override
    public CharSequence getTitle() {
        return mActionBarView.getTitle();
    }

    @Override
    public void setTitle(int resId) {
        mActionBarView.setTitle(resId);
    }

    @Override
    public void setTitle(CharSequence title) {
        mActionBarView.setTitle(title);
    }

    @Override
    public void setTitleGravity(int gravity) {
        mActionBarView.setTitleGravity(gravity);
    }

    @Override
    public int getTitleGravity() {
        return mActionBarView.getTitleGravity();
    }

    @Override
    public void setTitleVisibility(int visibility) {
        mActionBarView.setTitleVisibility(visibility);
    }

    @Override
    public int getTitleVisibility() {
        return mActionBarView.getTitleVisibility();
    }

    @Override
    public void setOnTitleClickListener(OnClickListener listener) {
        mActionBarView.setOnTitleClickListener(listener);
    }

    @Override
    public ActionMode startActionMode(Callback callback) {
        return mActionBarView.startActionMode(callback);
    }


    @Override
    public void stopActionMode() {
        mActionBarView.stopActionMode();

    }

    @Override
    public void enableRefresh(boolean enable) {
        mActionBarView.enableRefresh(enable);
    }

    @Override
    public void enableRefresh(boolean enable, int gravity) {
        mActionBarView.enableRefresh(enable,gravity);
    }

    @Override
    public void refreshing(boolean refresh) {
        mActionBarView.refreshing(refresh);
    }

    @Override
    public void setOnRefreshClickListener(OnClickListener listener) {
        mActionBarView.setOnRefreshClickListener(listener);
    }

    @Override
    public void setOnNavigationListener(OnNavigationListener onNavigationListener) {
        mActionBarView.setOnNavigationListener(onNavigationListener);
    }

    @Override
    public String[] getListNavigation() {
        return mActionBarView.getListNavigation();
    }

    @Override
    public void setListNavigation(String[] navs) {
        mActionBarView.setListNavigation(navs);
    }

    @Override
    public void clearListNavigation() {
        mActionBarView.clearListNavigation();
    }


    @Override
    public ActionMenu getActionMenu() {
        return mActionBarView.getActionMenu();
    }

    @Override
    public ArrayList<ActionMenuItem> getMenus() {
        return mActionBarView.getActions();
    }

    @Override
    public void setMenus(ArrayList<ActionMenuItem> actions) {
        mActionBarView.setActions(actions);
    }

    @Override
    public void clearActionMenus() {
        mActionBarView.clearActions();
    }

    @Override
    public ActionTab getActionTab() {
        return mActionBarView.getActionTab();
    }

    @Override
    public ArrayList<ActionTabItem> getTabs() {
        return mActionBarView.getTabs();
    }

    @Override
    public void setTabs(ArrayList<ActionTabItem> tabs) {
        mActionBarView.setTabs(tabs);
    }

    @Override
    public void clearActionTabs() {
        mActionBarView.setTitleVisibility(View.VISIBLE);
        mActionBarView.clearActionTabs();
    }

    @Override
    public void setCustomView(View view) {
        mActionBarView.setCustomView(view);
    }

    @Override
    public void removeCustomView() {
        mActionBarView.removeCustomView();
    }

    @Override
    public View findViewById(int id) {
        return mActionBarView.findViewById(id);
    }


}

File: actionbar/src/main/java/mobi/cangol/mobile/actionbar/internal/ActionBarImpl.kt
package mobi.cangol.mobile.actionbar.internal

import android.support.annotation.IdRes
import android.view.View
import android.view.View.OnClickListener
import mobi.cangol.mobile.actionbar.*
import mobi.cangol.mobile.actionbar.ActionMode.Callback
import mobi.cangol.mobile.actionbar.view.ActionBarView

/**
 * @author Cangol
 */
class ActionBarImpl(private val mActionBarView: ActionBarView) : ActionBar() {

    override var isShow: Boolean
        get() = mActionBarView.visibility == View.VISIBLE
        set(show) {
            mActionBarView.visibility = if (show) View.VISIBLE else View.GONE
        }

    override val title: CharSequence
        get() = mActionBarView.title

    override var titleGravity: Int
        get() = mActionBarView.titleGravity
        set(gravity) {
            mActionBarView.titleGravity = gravity
        }

    override var titleVisibility: Int
        get() = mActionBarView.titleVisibility
        set(visibility) {
            mActionBarView.titleVisibility = visibility
        }

    override var listNavigation: Array<String>?
        get() = mActionBarView.listNavigation
        set(navs) {
            mActionBarView.listNavigation = navs
        }


    override val actionMenu: ActionMenu
        get() = mActionBarView.actionMenu!!

    override var menus: MutableList<ActionMenuItem>
        get() = mActionBarView.actions
        set(actions) {
            mActionBarView.actions = actions
        }

    override val actionTab: ActionTab
        get() = mActionBarView.actionTab!!

    override var tabs: MutableList<ActionTabItem>
        get() = mActionBarView.tabs
        set(tabs) {
            mActionBarView.tabs = tabs
        }

    override fun setCustomHomeAsUpIndicator(homeId: Int, upId: Int) {
        mActionBarView.setCustomHomeAsUpIndicator(homeId, upId)
    }

    override fun resetCustomHomeAsUpIndicator() {
        mActionBarView.resetCustomHomeAsUpIndicator()
    }

    override fun setDisplayShowHomeEnabled(show: Boolean) {
        mActionBarView.setDisplayShowHomeEnabled(show)
    }

    override fun hideHomeAsUpIndicator() {
        mActionBarView.hideHomeAsUpIndicator()
    }

    override fun displayIndicator(slideOffset: Float) {
        mActionBarView.displayIndicator(slideOffset)
    }

    override fun displayHomeIndicator() {
        mActionBarView.displayHomeIndicator()
    }

    override fun displayUpIndicator() {
        mActionBarView.displayUpIndicator()
    }

    override fun setIndicatorColor(color: Int) {
        mActionBarView.setIndicatorColor(color)
    }

    override fun setLeftMenu(id: Int, text: Int, icon: Int, listener: OnClickListener) {
        mActionBarView.setLeftMenu(id, text, icon, listener)
    }

    override fun clearLeftMenu() {
        mActionBarView.clearLeftMenu()
    }

    override fun onBackPressed(): Boolean {
        return mActionBarView.onBackPressed()
    }

    override fun setBackgroundColor(color: Int) {
        mActionBarView.setBackgroundColor(color)
    }

    override fun setBackgroundResource(resId: Int) {
        mActionBarView.setBackgroundResource(resId)
    }

    override fun setTitle(resId: Int) {
        mActionBarView.setTitle(resId)
    }

    override fun setTitle(title: CharSequence) {
        mActionBarView.setTitle(title)
    }

    override fun setOnTitleClickListener(listener: OnClickListener) {
        mActionBarView.setOnTitleClickListener(listener)
    }

    override fun startActionMode(callback: Callback): ActionMode {
        return mActionBarView.startActionMode(callback)
    }


    override fun stopActionMode() {
        mActionBarView.stopActionMode()

    }

    override fun enableRefresh(enable: Boolean) {
        mActionBarView.enableRefresh(enable)
    }

    override fun enableRefresh(enable: Boolean, gravity: Int) {
        mActionBarView.enableRefresh(enable, gravity)
    }

    override fun refreshing(refresh: Boolean) {
        mActionBarView.refreshing(refresh)
    }

    override fun setOnRefreshClickListener(listener: OnClickListener) {
        mActionBarView.setOnRefreshClickListener(listener)
    }

    override fun setOnNavigationListener(onNavigationListener: OnNavigationListener) {
        mActionBarView.setOnNavigationListener(onNavigationListener)
    }

    override fun clearListNavigation() {
        mActionBarView.clearListNavigation()
    }

    override fun clearActionMenus() {
        mActionBarView.clearActions()
    }

    override fun clearActionTabs() {
        mActionBarView.titleVisibility = View.VISIBLE
        mActionBarView.clearActionTabs()
    }

    override fun setCustomView(view: View) {
        mActionBarView.setCustomView(view)
    }

    override fun removeCustomView() {
        mActionBarView.removeCustomView()
    }

    override fun <T : View> findViewById(@IdRes id: Int): T {
        return mActionBarView.findViewById(id)
    }
}